### PR TITLE
feat(framework) Add internal `flwr-clientapp` command

### PR DESCRIPTION
**Pull request by chongshenng**
### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/py/flwr/client/supernode/app.py [hunk 1]
```diff
@@ -119,6 +119,17 @@ def run_client_app() -> None:
     register_exit_handlers(event_type=EventType.RUN_CLIENT_APP_LEAVE)
 
 
+def exec_client_app() -> None:
```
Comment (tanertopal) [inline src/py/flwr/client/supernode/app.py:122]: As we use `exec` for the SuperExec I would prefer to reuse that word on the SuperNode side and just change all mentions of `exec_` to `start_` as in `start_client_app`

Comment (chongshenng) [reply src/py/flwr/client/supernode/app.py]: Updated to `flwr_clientapp`, as discussed.

File: src/py/flwr/client/supernode/app.py [hunk 2]
```diff
@@ -120,6 +120,17 @@ def run_client_app() -> None:
...
 
+def flwr_clientapp() -> None:
+    """Run process-isolated Flower client app."""
+    log(INFO, "Starting Flower ClientApp")
+
+    event(EventType.RUN_CLIENT_APP_ENTER)
+
+    # args = _parse_args_flwr_clientapp().parse_args()
+
+    register_exit_handlers(event_type=EventType.RUN_CLIENT_APP_LEAVE)
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:131]:
```suggestion
```

File: src/py/flwr/client/supernode/app.py [hunk 3]
```diff
@@ -283,6 +288,13 @@ def _parse_args_run_supernode() -> argparse.ArgumentParser:
         - `$HOME/.flwr/` in all other cases
     """,
     )
+    parser.add_argument(
+        "--isolate",
+        action="store_true",
+        help="Run the ClientApp in an isolated process from the SuperNode."
+        "In this mode, the ClientApp and SuperNode communicate via gRPC."
+        "By default, both SuperNode and ClientApp run in the same process.",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:296]:
```suggestion
        help="Run `ClientApp` in an isolated process from the SuperNode."
        "In this mode, `ClientApp` and SuperNode communicate via gRPC."
        "By default, both SuperNode and `ClientApp` run in the same process.",
```

File: src/py/flwr/client/supernode/app.py [hunk 4]
```diff
@@ -309,6 +321,23 @@ def _parse_args_run_client_app() -> argparse.ArgumentParser:
     return parser
 
 
+def _parse_args_flwr_clientapp() -> argparse.ArgumentParser:
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:324]: Let's use this directly in the `flwr_clientapp` function above and log the arguments that were passed

File: src/py/flwr/client/supernode/app.py [hunk 5]
```diff
@@ -283,6 +288,13 @@ def _parse_args_run_supernode() -> argparse.ArgumentParser:
         - `$HOME/.flwr/` in all other cases
     """,
     )
+    parser.add_argument(
+        "--isolate",
+        action="store_true",
+        help="Run `ClientApp` in an isolated process from the SuperNode."
+        "In this mode, `ClientApp` and SuperNode communicate via gRPC."
+        "By default, both SuperNode and `ClientApp` run in the same process.",
+    )
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:297]: @chongshenng this could be moved into a separate pr

File: src/py/flwr/client/supernode/app.py (hunk 4 above)
Comment (chongshenng) [reply src/py/flwr/client/supernode/app.py]: I've set the log level to `DEBUG`. Otherwise the outputs will be quite verbose for every round of `ClientApp` communication.

File: src/py/flwr/client/supernode/app.py [hunk 6]
```diff
@@ -120,6 +120,31 @@ def run_client_app() -> None:
     register_exit_handlers(event_type=EventType.RUN_CLIENT_APP_LEAVE)
 
 
+def flwr_clientapp() -> None:
+    """Run process-isolated Flower client app."""
+    log(INFO, "Starting Flower ClientApp")
+
+    parser = argparse.ArgumentParser(
+        description="Run a Flower client app",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:128]:
```suggestion
        description="Run a Flower ClientApp",
```

File: src/py/flwr/client/supernode/app.py [hunk 7]
```diff
@@ -120,6 +120,31 @@ def run_client_app() -> None:
     register_exit_handlers(event_type=EventType.RUN_CLIENT_APP_LEAVE)
 
 
+def flwr_clientapp() -> None:
+    """Run process-isolated Flower client app."""
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:124]:
```suggestion
    """Run process-isolated Flower ClientApp."""
```

File: src/py/flwr/client/supernode/app.py [hunk 8]
```diff
@@ -120,6 +120,31 @@ def run_client_app() -> None:
...
+def flwr_clientapp() -> None:
+    """Run process-isolated Flower ClientApp."""
+    log(INFO, "Starting Flower ClientApp")
+
+    parser = argparse.ArgumentParser(
+        description="Run a Flower ClientApp",
+    )
+    parser.add_argument(
+        "--address",
+        help="Address of SuperNode",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:132]:
```suggestion
        help="Address of SuperNode `ClientAppIo` gRPC servicer",
```

File: src/py/flwr/client/supernode/app.py [hunk 9]
```diff
@@ -120,6 +120,31 @@ def run_client_app() -> None:
...
+    parser = argparse.ArgumentParser(
+        description="Run a Flower ClientApp",
+    )
+    parser.add_argument(
+        "--address",
+        help="Address of SuperNode `ClientAppIo` gRPC servicer",
+    )
+    parser.add_argument(
+        "--token",
+        help="Unique token generated by SuperNode for each client app execution",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:136]:
```suggestion
        help="Unique token generated by SuperNode for each ClientApp execution",
```

File: src/py/flwr/client/supernode/app.py [hunk 10]
```diff
@@ -120,6 +120,31 @@ def run_client_app() -> None:
...
+        help="Address of SuperNode `ClientAppIo` gRPC servicer",
+    )
+    parser.add_argument(
+        "--token",
+        help="Unique token generated by SuperNode for each ClientApp execution",
+    )
+    args = parser.parse_args()
+    log(
+        DEBUG,
+        "Staring isolated `ClientApp` connected to SuperNode at %s "
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:141]:
```suggestion
        "Staring isolated `ClientApp` connected to SuperNode ClientAppIo at %s "
```

File: src/py/flwr/client/supernode/app.py [hunk 11]
```diff
@@ -120,6 +120,31 @@ def run_client_app() -> None:
...
+    )
+    parser.add_argument(
+        "--token",
+        help="Unique token generated by SuperNode for each ClientApp execution",
+    )
+    args = parser.parse_args()
+    log(
+        DEBUG,
+        "Staring isolated `ClientApp` connected to SuperNode ClientAppIo at %s "
+        "with the following token %s.",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:142]:
```suggestion
        "with the token %s",
```

File: src/py/flwr/client/supernode/app.py [hunk 12]
```diff
@@ -120,6 +120,31 @@ def run_client_app() -> None:
...
+def flwr_clientapp() -> None:
+    """Run process-isolated Flower ClientApp."""
+    log(INFO, "Starting Flower ClientApp")
+
+    parser = argparse.ArgumentParser(
+        description="Run a Flower ClientApp",
+    )
+    parser.add_argument(
+        "--address",
+        help="Address of SuperNode `ClientAppIo` gRPC servicer",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:132]:
```suggestion
        help="Address of SuperNode ClientAppIo gRPC servicer",
```

**Review by danieljanes — APPROVED** [review]
